resources/backend/sql_advanced.md

```diff
@@ -1,9 +1,22 @@
 # SQL Advanced topics
 
 ## What is a GROUP BY?
+A [SQL GROUP BY](https://www.w3schools.com/sql/sql_groupby.asp) is a statement used to group rows with the same values together.
+In this codebase, GROUP BY can be used to select accounts, where the user can group the results by the account ID:
+```mysql
+SELECT accounts.*, SUM(amount) FROM accounts JOIN transactions ON accounts.id=transactions.account_id GROUP BY account_id HAVING SUM(amount) < 0
+```
+As seen above, GROUP BY is commonly used with aggregate functions (e.g. SUM, MAX, MIN).
+
 ## What is HAVING?
+[SQL HAVING](https://www.w3schools.com/sql/sql_having.asp) is a statement used where the WHERE keyword cannot be used with aggregate functions (e.g. SUM, MAX, MIN). For example, in this codebase, HAVING is used to select accounts with negative balances:
+```mysql
+SELECT accounts.*, SUM(amount) FROM accounts JOIN transactions ON accounts.id=transactions.account_id GROUP BY account_id HAVING SUM(amount) < 0
+```
```

Comment:
Do you think you can include the results that you get? In other words, basically we expect x columns. The first (x-1) columns are the account columns and the last column is the SUM of blah balh